### PR TITLE
Fix/revpi 1836/fix pinctrl

### PR DESCRIPTION
**Pull request by linosanfilippo-kunbus**

### CODE REVIEW

File: arch/arm/boot/dts/overlays/revpi-compact-overlay.dts [hunk 1]
```diff
@@ -103,11 +103,6 @@
 				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
 				brcm,pull     = <BCM2835_PUD_OFF>;
 			};
-			eth1_int_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-compact-overlay.dts:106]: Why don't you add this pinmux to the eth1 node?

File: arch/arm/boot/dts/overlays/revpi-compact-overlay.dts [hunk 2]
```diff
@@ -120,12 +115,6 @@
 				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
 				brcm,pull     = <BCM2835_PUD_OFF>;
 			};
-			ain_mux_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-compact-overlay.dts:123]: Why don't you add this pinmux to the ain_mux node?

File: arch/arm/boot/dts/overlays/revpi-compact-overlay.dts [hunk 3]
```diff
@@ -140,12 +129,6 @@
 						 BCM2835_FSEL_GPIO_IN>;
 				brcm,pull     = <BCM2835_PUD_OFF>;
 			};
-			leds_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-compact-overlay.dts:143]: And the same as above but for the leds node?

File: arch/arm/boot/dts/overlays/revpi-core-overlay.dts [hunk 4]
```diff
@@ -80,12 +80,6 @@
 				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
 				brcm,pull     = <BCM2835_PUD_OFF>;
 			};
-			eth1_2_reset_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-core-overlay.dts:83]: The this pin is used by the pb_rst node. So it should reference this pin!

File: arch/arm/boot/dts/overlays/revpi-flat-overlay.dts [hunk 5]
```diff
@@ -117,30 +117,12 @@
 				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
 				brcm,pull     = <BCM2835_PUD_OFF>;
 			};
-			tpm_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-flat-overlay.dts:120]: The irq pin is used by the tpm node. You should add a reference from there to this pinctrol. The reset is currently unused by the tpm node. If the driver supports a reset gpio we could add it here.

File: arch/arm/boot/dts/overlays/revpi-flat-overlay.dts [hunk 6]
```diff
@@ -117,30 +117,12 @@
...
-				/* irq reset */
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-flat-overlay.dts:121]: The order of the pins in the comment is wrong!
pin 0 is the reset
pin 1 is the irq

File: arch/arm/boot/dts/overlays/revpi-flat-overlay.dts [hunk 7]
```diff
@@ -117,30 +117,12 @@
...
-				brcm,pins     = <0 1>;
-				brcm,function = <BCM2835_FSEL_GPIO_OUT
-						 BCM2835_FSEL_GPIO_IN>;
-				brcm,pull     = <BCM2835_PUD_OFF>;
-			};
-			mbus_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-flat-overlay.dts:127]: The mbus node uses pin 21 as interrupt. It should reference this pinctrl

File: arch/arm/boot/dts/overlays/revpi-flat-overlay.dts [hunk 8]
```diff
@@ -117,30 +117,12 @@
...
-				brcm,function = <BCM2835_FSEL_GPIO_IN>;
-				brcm,pull     = <BCM2835_PUD_OFF>;
-			};
 			watchdog_gpio4: watchdog_pins {
 				/* enable ping */
 				brcm,pins     = <4 5>;
 				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
 				brcm,pull     = <BCM2835_PUD_OFF>;
 			};
-			button_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-flat-overlay.dts:139]: The gpio-keys should reference this pinctrol

File: arch/arm/boot/dts/overlays/revpi-connect-overlay.dts [hunk 9]
```diff
@@ -85,21 +85,6 @@
 				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
 				brcm,pull     = <BCM2835_PUD_OFF>;
 			};
-			eth2_int_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-connect-overlay.dts:88]: The pileft node should reference this pinctrol

File: arch/arm/boot/dts/overlays/revpi-connect-overlay.dts [hunk 10]
```diff
@@ -85,21 +85,6 @@
...
-				brcm,pins     = <8>;
-				brcm,function = <BCM2835_FSEL_GPIO_IN>;
-				brcm,pull     = <BCM2835_PUD_OFF>;
-			};
-			eth2_reset_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-connect-overlay.dts:93]: The pb_rst node should reference this pinctrl

File: arch/arm/boot/dts/overlays/revpi-connect-overlay.dts [hunk 11]
```diff
@@ -85,21 +85,6 @@
...
-				brcm,pins     = <40>;
-				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
-				brcm,pull     = <BCM2835_PUD_OFF>;
-			};
-			conbridge_int_pins {
```
Comment (iluminat23) [inline arch/arm/boot/dts/overlays/revpi-connect-overlay.dts:98]: Add this to the pincontroller or move it to https://github.com/RevolutionPi/linux/blob/devel/revpi-5.10/arch/arm/boot/dts/overlays/revpi-con-can-overlay.dts as it is used there.

**Review by iluminat23 — CHANGES_REQUESTED** [review]

**Review by iluminat23 — COMMENTED** [review]
Have checked if there is any difference in behavior with these changes?

**Review by iluminat23 — DISMISSED** [review]
Looks good to me

**Comment by l1k** [comment]
I think by convention, the `compatible` property always comes first. The commits in this pull don't handle that in a uniform manner, e.g. in the Connect overlay, `compatible` precedes the `pinctrl` properties, wheras in the Compact overlay, it's the other way round. The former seems more correct to me. In any case it would be nice to have the same order across all overlays.

**Review by l1k — CHANGES_REQUESTED** [review]
_

**Comment by linosanfilippo-kunbus** [comment]
I agree, I will adjust it to have to compatible first everywhere

File: arch/arm/boot/dts/overlays/revpi-flat-overlay.dts (hunk 6 above)
Comment (iluminat23) [reply arch/arm/boot/dts/overlays/revpi-flat-overlay.dts:121]: Could you adjust the comment here, while you are on it?

**Review by iluminat23 — CHANGES_REQUESTED** [review]
While you are on it could you fix the comment of the tpm_pins irq and reset are reversed.

**Comment by linosanfilippo-kunbus** [comment]
Sure, no prob

**Review by iluminat23 — APPROVED** [review]

**Review by l1k — APPROVED** [review]

**Pull request merged**